.github/workflows/binaries.yml

```diff
@@ -0,0 +1,114 @@
+name: Build binaries
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'version to label binaries'
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'version to label binaries'
+        required: true
+        type: string
+
+jobs:
+  build:
+    permissions:
+      contents: read
+      packages: read
+    runs-on: ${{ matrix.build.os }}
+    strategy:
+      matrix:
+        build:
+          - target: aarch64-apple-darwin
+            os: macos-12
+          - target: x86_64-apple-darwin
+            os: macos-12
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-22.04
+          - target: aarch64-unknown-linux-gnu
+            os: ubuntu-22.04
+    env:
+      RUST_BACKTRACE: full
+      RUSTC_WRAPPER: "sccache"
+      SCCACHE_GHA_ENABLED: "true"
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: ./.github/actions/clean-runner
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          # need to provide a toolchain version because this action cannot read rust-toolchain.toml. See https://github.com/actions-rs/toolchain/issues/126
+          toolchain: stable
+
+      # necessary because the toolchain@v1 action would install this target for stable only
+      - name: Install aarch64-apple-darwin Rust toolchain
+        if: ${{ matrix.build.target == 'aarch64-apple-darwin' }}
+        run: rustup target add aarch64-apple-darwin
+
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.3
+
+      - name: Cache Rust
+        uses: Swatinem/rust-cache@v2
+
+      - name: Install cargo-license
+        run: cargo install cargo-license
+
+      - name: Install protoc
+        if: ${{ matrix.build.os == 'macos-12' }} # installed in cross docker image in linux builds
+        uses: ./.github/actions/install-protoc
+
+      - name: Setup just
+        uses: extractions/setup-just@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Generate notice
+        run: just notice-file
+
+      - name: Compile Mac binaries
+        if: ${{ matrix.build.os == 'macos-12' }}
+        uses: actions-rs/cargo@v1
+        env:
+          MACOSX_DEPLOYMENT_TARGET: 10.14.0
+          CC: sccache clang -target ${{ matrix.build.target }}
+          CXX: sccache clang++ -target ${{ matrix.build.target }}
```

Comment:
Is this needed when building on macos? Differently asked, can the build step for linux and mac be exactly the same?

Reply:
this flag must be set for mac cross compile to work, otherwise librdkafka will get built for amd64. on linux runner, gcc is used

Reply:
Maybe add this as a comment for future Restate generations.

Reply:
Interesting. I am wondering why I can cross compile on my machine for x86 darwin because I haven't explicitly configured the target for clang and have not set up clang++ (at least I cannot remember).

Reply:
if we didnt want to use sccache we might be able to get away without these flags, but once I added the sccache prefix it didnt work without me setting -target. These things are very unpredictable...

Reply:
it may also be related to cmake/non-cmake builds; cmake made it vary a bit in outcome from environment to environment. without cmake, we have predictable outcomes, but we sometimes need to set flags